File: native/ios/Sources/Theme.swift
import SwiftUI

/// QRME dark-OLED palette, shared across every screen.
enum Theme {
    static let scrTop   = Color(hex: 0x0E1626)
    static let scrBot   = Color(hex: 0x0A0F1C)
    static let card     = Color(hex: 0x182238)
    static let line     = Color(hex: 0x26314E)
    static let txt      = Color(hex: 0xEEF1F7)
    static let t2       = Color(hex: 0x8A94AD)
    static let t3       = Color(hex: 0x626D88)
    static let brandA   = Color(hex: 0x7C5CFF)
    static let brandB   = Color(hex: 0x3AA0FF)
    static let green    = Color(hex: 0x43E08A)
    static let red      = Color(hex: 0xFF3B30)
    static let amber    = Color(hex: 0xF7B731)

    static let bg = LinearGradient(colors: [scrTop, scrBot],
                                   startPoint: .top, endPoint: .bottom)
    static let brand = LinearGradient(colors: [brandA, brandB],
                                      startPoint: .topLeading, endPoint: .bottomTrailing)
}

extension Color {
    init(hex: UInt32) {
        self.init(.sRGB,
                  red: Double((hex >> 16) & 0xFF) / 255,
                  green: Double((hex >> 8) & 0xFF) / 255,
                  blue: Double(hex & 0xFF) / 255,
                  opacity: 1)
    }
}

/// A rounded, hairline-bordered surface used for every card on every screen.
struct CardBackground: ViewModifier {
    func body(content: Content) -> some View {
        content
            .padding(16)
            .background(Theme.card.opacity(0.9))
            .overlay(RoundedRectangle(cornerRadius: 16).stroke(Theme.line, lineWidth: 1))
            .clipShape(RoundedRectangle(cornerRadius: 16))
    }
}

extension View {
    func card() -> some View { modifier(CardBackground()) }
}

File: native/android/app/src/main/java/app/qrme/studio/ui/Theme.kt
package app.qrme.studio.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Typography
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/** QRME dark-OLED palette. */
object Qrme {
    val ScrTop = Color(0xFF0E1626)
    val ScrBot = Color(0xFF0A0F1C)
    val Card = Color(0xFF182238)
    val Line = Color(0xFF26314E)
    val Txt = Color(0xFFEEF1F7)
    val T2 = Color(0xFF8A94AD)
    val T3 = Color(0xFF626D88)
    val BrandA = Color(0xFF7C5CFF)
    val BrandB = Color(0xFF3AA0FF)
    val Green = Color(0xFF43E08A)
    val Red = Color(0xFFFF3B30)
    val Amber = Color(0xFFF7B731)
    val Brand = Brush.linearGradient(listOf(BrandA, BrandB))
    val Bg = Brush.verticalGradient(listOf(ScrTop, ScrBot))
}

private val QrmeColors = darkColorScheme(
    primary = Qrme.BrandA,
    background = Qrme.ScrBot,
    surface = Qrme.Card,
    onPrimary = Color.White,
    onBackground = Qrme.Txt,
    onSurface = Qrme.Txt,
)

@Composable
fun QrmeTheme(content: @Composable () -> Unit) {
    MaterialTheme(colorScheme = QrmeColors, typography = Typography()) {
        Surface(color = Qrme.ScrBot, content = content)
    }
}

/** Rounded, hairline-bordered card surface used across every screen. */
fun Modifier.card(): Modifier =
    this
        .clip(RoundedCornerShape(16.dp))
        .background(Qrme.Card.copy(alpha = 0.9f))
        .border(1.dp, Qrme.Line, RoundedCornerShape(16.dp))
        .padding(16.dp)
